scrapy/pipelines/media.py

```diff
@@ -16,6 +19,7 @@
 class MediaPipeline(object):
 
     LOG_FAILED_RESULTS = True
+    ALLOW_REDIRECTS = False
```

Comment:
@RabidCicada , you'll probably need to clone https://github.com/Granitosaurus/scrapy/tree/mediapipeline_redirect_fix_2004 in any case.
Then you can submit a pull request to than branch in @Granitosaurus 's repo,
or submit a new PR (with @Granitosaurus work + your commit) against scrapy master